Xamarin.Forms.Core/Page.cs

```diff
@@ -317,6 +317,8 @@ public void SendAppearing()
 
 			var pageContainer = this as IPageContainer<Page>;
 			pageContainer?.CurrentPage?.SendAppearing();
+
+			Application.Current?.OnPageAppearing(this);
```

Comment:
When you have a page and want the application, use Page.parent.parent.parent blah blah blah until you find the Application or strike null. If you strike null there is no applciation because you are either not parented (in which case something really fucky has happened) or you are inside a pre-Application based Forms app in which case you shouldn't trigger those events anyway.

Reply:
Fixed! Thanks 👍